src/alg/eddsa_compact.rs

```diff
@@ -0,0 +1,131 @@
+use ed25519_compact::{KeyPair, PublicKey, SecretKey, Seed, Signature};
+use rand::Rng;
+use rand_core::{CryptoRng, RngCore};
+
+use std::borrow::Cow;
+use std::fmt;
+
+use crate::{Algorithm, AlgorithmSignature, Renamed};
+
+impl AlgorithmSignature for Signature {
+    fn try_from_slice(bytes: &[u8]) -> anyhow::Result<Self> {
+        let mut signature = [0u8; Signature::BYTES];
+        if bytes.len() != signature.len() {
+            return Err(ed25519_compact::Error::SignatureMismatch.into());
+        }
+        signature.copy_from_slice(bytes);
+        Ok(Self::new(signature))
+    }
+
+    fn as_bytes(&self) -> Cow<[u8]> {
+        Cow::Owned(self.as_ref().to_vec())
+    }
+}
+
+/// A verification key.
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct Ed25519VerifyingKey(PublicKey);
+
+impl AsRef<PublicKey> for Ed25519VerifyingKey {
+    fn as_ref(&self) -> &PublicKey {
+        &self.0
+    }
+}
+
+impl Ed25519VerifyingKey {
+    /// Create a verification key from a slice.
+    pub fn from_slice(raw: &[u8]) -> anyhow::Result<Ed25519VerifyingKey> {
+        Ok(Ed25519VerifyingKey(PublicKey::from_slice(raw)?))
+    }
+
+    /// Return the key as raw bytes.
+    pub fn as_bytes(&self) -> Cow<[u8]> {
+        Cow::Owned(self.as_ref().as_ref().to_vec())
+    }
+}
+
+/// A signing key.
+pub struct Ed25519SigningKey(SecretKey);
```

Comment:
I'm not entirely sure I understand the reasoning behind newtypes; see https://github.com/slowli/jwt-compact/issues/12#issuecomment-660537106.